queue/src/fiskaltrust.Middleware.Storage.AzureTableStorage/Repositories/AzureTableStorageQueueItemRepository.cs

```diff
@@ -97,15 +97,30 @@ public override async Task InsertAsync(ftQueueItem storageEntity)
             await _receiptReferenceIndexRepository.InsertAsync(new ReceiptReferenceIndex { cbReceiptReference = storageEntity.cbReceiptReference, ftQueueItemId = storageEntity.ftQueueItemId });
         }
 
-        public override async Task InsertOrUpdateAsync(ftQueueItem storageEntity)
+
+        public async Task InsertOrUpdateAsync(ftQueueItem storageEntity)
         {
-            await base.InsertOrUpdateAsync(storageEntity);
+            EntityUpdated(storageEntity);
+            var entity = MapToAzureEntity(storageEntity);
+            await _tableClient.UpsertEntityAsync(entity, TableUpdateMode.Replace);
             await _receiptReferenceIndexRepository.InsertOrUpdateAsync(new ReceiptReferenceIndex { cbReceiptReference = storageEntity.cbReceiptReference, ftQueueItemId = storageEntity.ftQueueItemId });
         }
 
+        public IAsyncEnumerable<ftQueueItem> GetByTimeStampRangeAsync(long fromInclusive, long toInclusive)
+        {
+            var result = _tableClient.QueryAsync<TableEntity>(filter: TableClient.CreateQueryFilter($"TimeStamp ge {fromInclusive} and TimeStamp le {toInclusive}"));
+            return result.Select(MapToStorageEntity);
```

Comment:
Another thing that we should double check is if the order that is being returned is correct, or if we need to order differently.